### ISSUE TITLE
After Update to 2.6.0 Every Time I Open the Page with Elementor, the Page Says It's a Draft

### ISSUE THREAD

**Issue by black-eye** [elementor/elementor#8510]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description
The title says it quite clearly. Even if I update the "draft" and open it again, it says the same again. The changes seem to be saved, though.

![obrazek](https://user-images.githubusercontent.com/5634870/60926569-3b525380-a2a7-11e9-9669-70e5e32985e4.png)

It is worth mentioning that I limit the number of revisions by this code:

```
function bleye_control_revisions($num, $post) {
  $num = 20;
  return $num;
}
add_filter( 'wp_revisions_to_keep', 'bleye_control_revisions', 20, 2 );
```

But all worked correctly before I updated to last versions of E and EPro.

## Steps to reproduce
As Written above.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: mariadb.org binary distribution v10.3.15
	PHP Version: 7.2.19
	PHP Max Input Vars: 5000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.2.2
	Site URL: https://www.wushu.cz/prac
	Home URL: https://www.wushu.cz/prac
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 256M
	Permalink Structure: /clanky/%postname%/
	Language: cs-CZ
	Timezone: Europe/Prague
	Debug Mode: Inactive

== Theme ==
	Name: Wushu WPBF
	Version: 1.0.0
	Author: Petr Jandák
	Child Theme: Yes
	Parent Theme Name: Page Builder Framework
	Parent Theme Version: 2.0.3
	Parent Theme Author: David Vongries

== User ==
	Role: administrator
	WP Profile lang: cs_CZ
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:67.0) Gecko/20100101 Firefox/67.0

== Active Plugins ==
	Activity Log
		Version: 2.5.2
		Author: Activity Log Team

	Admin Menu Editor
		Version: 1.9
		Author: Janis Elsts

	Advanced Custom Fields PRO
		Version: 5.7.13
		Author: Elliot Condon

	All-in-One WP Migration
		Version: 6.77
		Author: ServMask

	Caldera Forms
		Version: 1.7.6
		Author: Caldera Labs

	Central Color Palette
		Version: 1.13.10
		Author: Daniel Schneider

	Classic Editor
		Version: 1.5
		Author: WordPress Contributors

	Clean Image Filenames
		Version: 1.2.1
		Author: Upperdog

	Cookie Notice
		Version: 1.2.46
		Author: dFactory

	Disable Comments
		Version: 1.9.0
		Author: Samir Shah

	Duplicate Post
		Version: 3.2.2
		Author: Enrico Battocchi

	DynamicConditions
		Version: 1.2.1
		Author: RTO GmbH

	Elementor
		Version: 2.6.0
		Author: Elementor.com

	Elementor Extras
		Version: 2.1.4
		Author: Namogo

	Elementor Pro
		Version: 2.5.11
		Author: Elementor.com

	Enable Media Replace
		Version: 3.2.9
		Author: ShortPixel

	Intuitive Custom Post Order
		Version: 3.1.2
		Author: hijiri

	iThemes Security
		Version: 7.3.3
		Author: iThemes

	Lightbox with PhotoSwipe
		Version: 1.99
		Author: Arno Welzel

	Nested Pages
		Version: 3.0.11
		Author: Kyle Phillips

	Page Builder Framework Premium Addon
		Version: 2.0.2
		Author: David Vongries

	Piotnet Addons For Elementor Pro
		Version: 4.2.9
		Author: Luong Huu Phuoc (Louis Hufer)

	ShortPixel Image Optimizer
		Version: 4.13.1
		Author: ShortPixel

	Simple 301 Redirects
		Version: 1.07
		Author: Scott Nellé

	The SEO Framework
		Version: 3.2.4
		Author: Sybre Waaijer

	TinyMCE Advanced
		Version: 5.2.1
		Author: Andrew Ozz

	WordPress Importer
		Version: 0.6.4
		Author: wordpressdotorg

	WP Mail SMTP
		Version: 1.4.2
		Author: WPForms

	WP Media folder
		Version: 4.7.12
		Author: Joomunited

	Zalomení
		Version: 1.4.7
		Author: Honza Skypala


== Log ==
	: 
Log: showing 20 of 342019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_4_products Start  
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_4_products Finished 
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_0_sitemap Start  
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_0_sitemap Finished 
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_0_popup_border_radius Start  
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_0_popup_border_radius Finished 
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_4_form Start  
2019-04-05 21:52:54 [info] Elementor Pro/Upgrades - _v_2_5_4_form Finished 
2019-04-05 21:52:54 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.4.5',
  'to' => '2.5.4',
)]
2019-04-16 13:29:15 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.11',
  'to' => '2.5.14',
)]
2019-04-16 13:29:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.4',
  'to' => '2.5.5',
)]
2019-05-02 16:48:49 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.5',
  'to' => '2.5.6',
)]
2019-05-11 02:49:32 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.14',
  'to' => '2.5.15',
)]
2019-05-11 02:49:58 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.6',
  'to' => '2.5.8',
)]
2019-07-09 23:09:53 [info] elementor::elementor_updater Started 
2019-07-09 23:09:53 [info] Elementor/Upgrades - _v_2_6_0_fa4_migration_flag Start  
2019-07-09 23:09:53 [info] Elementor/Upgrades - _v_2_6_0_fa4_migration_flag Finished 
2019-07-09 23:09:53 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.15',
  'to' => '2.6.0',
)]
2019-07-09 23:09:53 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.15',
  'to' => '2.6.0',
)]
2019-07-09 23:10:49 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.8',
  'to' => '2.5.11',
)]

JS: showing 7 of 7JS: 2019-04-05 22:23:23 [error X 10][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.11:2:179981] this._parent is undefined 
JS: 2019-04-16 13:09:14 [error X 21][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.14:2:75462] setting getter-only property \"type\" 
JS: 2019-04-27 20:48:17 [error X 1][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.14:2:8492] this.ui.select is undefined 
JS: 2019-05-01 19:09:14 [error X 11][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.14:2:53462] this.ui.sliders.filter(...)[0] is undefined 
JS: 2019-05-03 17:29:25 [error X 1][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=2.5.14:2:0]  
JS: 2019-05-06 15:04:19 [error X 6][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/js/common.min.js?ver=2.5.14:2:10917] Object.entries is not a function 
JS: 2019-05-06 15:04:41 [error X 2][https://www.wushu.cz/prac/wp-content/plugins/elementor/assets/lib/inline-editor/js/inline-editor.min.js?ver=2.5.14:1:3265] Object.values is not a function 

PHP: showing 2 of 2PHP: 2019-04-14 12:37:56 [warning X 160][/www/doc/www.wushu.cz/www/prac/wp-content/plugins/elementor-extras/modules/gallery/widgets/gallery.php::1685] Invalid argument supplied for foreach() [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-05-09 17:55:53 [notice X 5][/www/doc/www.wushu.cz/www/prac/wp-content/plugins/elementor-extras/modules/gallery/widgets/gallery.php::1937] Trying to get property 'post_excerpt' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]


				
```
</details>

**Comment by inaffect-ag** [comment]
I have the same problem. Editing new content is no longer possible with the current version. Please find a quick solution!

**Comment by joedaniels** [comment]
Yep same issue here

**Comment by vaka2vaka** [comment]
Yep same issue here

**Comment by inaffect-ag** [comment]
![image](https://user-images.githubusercontent.com/6585358/60968422-4f389c80-a31d-11e9-9142-0c9854bd2910.png)

The following error is displayed in the console.

**Comment by black-eye** [comment]
@inaffect-ag Not sure, the console error is related to this issue. 
In my case, I have no errors in the console.

**Comment by inaffect-ag** [comment]
@black-eye I tested it on an installation where I didn't update Elementor. I completely uninstalled the plugin and installed the latest version again. It worked fine during this process. I assume that the error occurs during the automatic update process. Unfortunately, this solution does not work if you have already run the automatic update process.

**Comment by black-eye** [comment]
@inaffect-ag Hmm, interesting. On my site I've also done the update process from 2.5.something and no error ocured. Hopefuly this info helps developers to resolve the problem.

**Comment by evgeniyshay** [comment]
Error in this file
https://github.com/elementor/elementor/blob/master/assets/dev/js/editor/editor.js
in checkPageStatus function
line: 755
`if ( elementor.config.current_revision_id !== elementor.config.document.id ) {`
the condition is triggered every time because
`typeof elementor.config.current_revision_id` return string
and
`typeof elementor.config.document.id` return number.
That's why with strict comparison operator '!==' it always return true, even though the post id is matched.

**Comment by inaffect-ag** [comment]
@evgeniyshay Can we correct this ourselves?

**Comment by inaffect-ag** [comment]
@evgeniyshay I commented out the block and ran a test. It had no effect on the editing options.

**Comment by evgeniyshay** [comment]
@inaffect-ag you need replace this line
`if ( elementor.config.current_revision_id !== elementor.config.document.id ) {`
with this
`if ( Number( elementor.config.current_revision_id ) !== elementor.config.document.id ) {`
and compile js with npm 
I hope it helps

**Comment by inaffect-ag** [comment]
@evgeniyshay I have tested it, the dialogue can indeed be removed in this way. Unfortunately, editing is still not possible. The loading process does not interrupt.

**Comment by fedeoasis** [comment]
Same issue, but I can confirm that changes are saved.

**Comment by TheMangoTrain** [comment]
I also have this issue for 2 days now.

... and are the widgets loading for you guys?

For me, widgets area is a white haze with a loading animation.

**Comment by black-eye** [comment]
For me widget area worked well.

**Comment by Zahtar** [comment]
Have this problem on two sites now after updating - can edit and save, but attempting to preview the page from within the elementor editor gives the WP error "Sorry, you are not allowed to preview drafts."

**Comment by inaffect-ag** [comment]
@TheMangoTrain I behaved exactly the same way.

**Comment by inaffect-ag** [comment]
@arielk When can we expect an update here? For us as an agency this situation is very unpleasant.

**Comment by ludyamano** [comment]
@inaffect-ag same issue here. I cannot edit or create a page using Elementor right now. I cannot drag element to page anymore. There's a loading animation in front of it.

![Captura de Tela 2019-07-13 às 10 30 44](https://user-images.githubusercontent.com/55014/61172348-59df7580-a559-11e9-8ac2-fd54d577f956.png)

**Comment by ludyamano** [comment]
After reviewing my code, i realized i was using a custom widget that i've created that was extending the Posts_Base. The problem was the Related Posts Group Control in ``register_query_section_controls()``. 

Everything went back to normal after removing the following code:
<pre>
`
		$this->add_group_control(
			Group_Control_Related::get_type(),
			[
				'name' => 'mirago-presential',
				'presets' => [ 'full' ],
				'exclude' => [
					'posts_per_page', //use the one from Layout section
				],
			]
		);
`
</pre>

**Comment by shilo-ey** [comment]
Hi all, 

Please note that this issue relates to the "draft" message ONLY.

The spinner compression error is temporarily resolved in Elementor v2.6.2
Please keep this issue just for the "draft" issue.

We are investigating this issue and will resolve it soon.

Thanks!
